BugDoom - web app/src/main/java/com/infoshareacademy/pl/service/TrackService.java

```diff
@@ -1,46 +1,50 @@
 package com.infoshareacademy.pl.service;
 
+import com.infoshareacademy.pl.exception.TrackNotFoundException;
 import com.infoshareacademy.pl.model.Track;
 import com.infoshareacademy.pl.repository.TrackRepository;
-import com.infoshareacademy.pl.repository.UserRepository;
 import org.springframework.stereotype.Service;
 
+
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.NoSuchElementException;
+import java.util.Random;
 
 @Service
 public class TrackService implements TrackRepository {
-    private final String TRACK_FILE_PATH = FilePathConstants.TRACK_FILE_PATH;
+
+    private static final String TRACK_FILE_PATH = FilePathConstants.TRACK_FILE_PATH;
     private final DataService<Track> dataService = new DataService<>();
 
+    @Override
     public List<Track> getAllTracks() throws IOException {
         return new ArrayList<>(dataService.readFromFile(TRACK_FILE_PATH, Track[].class));
     }
 
-    public void addTrackToFile(Track trackToAdd) throws IOException {
+    public void addTrack(Track trackToAdd) throws IOException {
         List<Track> allTracks = getAllTracks();
         allTracks.add(trackToAdd);
         saveTracksToFile(allTracks);
     }
 
-    public void deleteTrack(Track trackToDelete) throws IOException {
+    public void removeTrackById(long trackId) throws IOException {
         List<Track> allTracks = getAllTracks();
+        Track trackToDelete = findTrackById(trackId);
         allTracks.remove(trackToDelete);
         saveTracksToFile(allTracks);
     }
 
     private void saveTracksToFile(List<Track> tracksToSave) throws IOException {
         dataService.saveToFile(tracksToSave, TRACK_FILE_PATH);
     }
-
-    @Override
-    public Track findTrackById(long trackId) throws NoSuchElementException, IOException {
-        List<Track> allTracks = getAllTracks();
-        return allTracks.stream()
-                .filter(t -> t.getTrackId() == (trackId))
-                .findFirst()
-                .orElseThrow();
+@Override
+    public Track findTrackById(long trackId) throws IOException{
+        List<Track>allTracks = getAllTracks();
+        return allTracks.stream().filter(track -> track.getTrackId() == trackId)
+                .findFirst().orElseThrow(() -> new TrackNotFoundException("Nie ma trasy o takim ID: %s".formatted(trackId)));
+    }
+    public long createRandomId() {
+        return new Random().nextLong(1000);
```

Comment:
warto rozdzielić trochę ten kod, żeby .filter(...), .findFirst() oraz .orElseThrow(...) były w osobnych linijkach dla większej czytelności

Reply:
to już chyba indywidualne podejście :)